Userland/Libraries/LibWeb/Painting/RecordingPainter.cpp

```diff
@@ -179,7 +179,8 @@ void RecordingPainter::draw_scaled_bitmap(Gfx::IntRect const& dst_rect, Gfx::Bit
 {
     push_command(DrawScaledBitmap {
         .dst_rect = state().translation.map(dst_rect),
-        .bitmap = bitmap,
+        // NOTE: Bitmap has to be cloned to prevent concurrent access from main and rendering threads.
+        .bitmap = MUST(bitmap.clone()),
```

Comment:
I wonder if we could have some Refcounting-assisted-COW mechanism here instead, as some images might not actually change in between frames
(assuming this is the general image rendering path)

Reply:
Not a thing that needs to be done now, if it is worth it

Reply:
COW definitely seems better than clone!

Reply:
I profiled pages with canvas to make sure it does not result in significant perf regression -- bitmap cloning was not visible in profiles. I think we have many other perf problems to solve until implementing COW will be reasonable :)